renormalizer/mps/gs.py

```diff
@@ -104,8 +103,28 @@ def optimize_mps(mps: Mps, mpo: Mpo):
 
 
 def optimize_mps_dmrg(mps, mpo):
-    """
-    1 or 2 site optimization procedure
+    r""" DMRG ground state algorithm and state-averaged excited states algorithm
+    
+    Parameters
+    ----------
+    mps : renormalizer.mps.Mps
+        initial guess of mps
+    mpo : renormalizer.mps.Mpo 
+        mpo of Hamiltonian
+    
+    Returns
+    -------
+    energy : list
+        list of energy of each marco sweep.
+        :math:`[e_0, e_0, \cdots, e_0]` if `nroots=1`.
+        :math:`[[e_0, \cdots, e_n], \dots, [e_0, \cdots, e_n]]` if `nroots=n`.
+    mps : renormalizer.mps.Mps
```

Comment:
Is it necessary to return the mps? The API gives me the impression that the modification is not in-place. If this is for development convenience, the behavior(in-place modification) should be documented.

Reply:
I think an optimized mps should be returned although the input mps is updated inplace like in `mp.py`, many methods return `self`. I will add some docs.